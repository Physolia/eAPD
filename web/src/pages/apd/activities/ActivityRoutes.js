import PropTypes from 'prop-types';
import React from 'react';
import {
  Switch,
  Route,
  useRouteMatch as actualUseRouteMatch
} from 'react-router-dom';

import ContractorResources from '../../../containers/activity/ContractorResources';
import CostAllocation from '../../../containers/activity/CostAllocation';
import FFP from '../../../containers/activity/CostAllocateFFP';
import Costs from '../../../containers/activity/Costs';
import Milestones from '../../../containers/activity/Milestones';
import Overview from '../../../containers/activity/Overview';
import Outcomes from '../../../containers/activity/Outcomes';
import StandardsAndConditions from '../../../containers/activity/StandardsAndConditions';
import { Section } from '../../../components/Section';

const ActivityRoutes = ({ activityIndex, useRouteMatch }) => {
  const { path } = useRouteMatch();

  return (
    <Switch>
      <Route path={`${path}/oms`}>
        <Section>
          <Outcomes activityIndex={activityIndex} />
          <Milestones activityIndex={activityIndex} />
        </Section>
      </Route>
      <Route path={`${path}/state-costs`}>
        <Section>
          <Costs activityIndex={activityIndex} />
        </Section>
      </Route>
      <Route path={`${path}/contractor-costs`}>
        <Section>
          <ContractorResources activityIndex={activityIndex} />
        </Section>
      </Route>
      <Route path={`${path}/cost-allocation`}>
        <Section>
          <CostAllocation activityIndex={activityIndex} />
        </Section>
      </Route>
      <Route path={`${path}/ffp`}>
        <Section>
          <FFP activityIndex={activityIndex} />
        </Section>
      </Route>
      <Route>
        <Section>
          <Overview activityIndex={activityIndex} />
          <StandardsAndConditions activityIndex={activityIndex} />
        </Section>
      </Route>
    </Switch>
  );
};

ActivityRoutes.defaultProps = {
  useRouteMatch: actualUseRouteMatch
};

ActivityRoutes.propTypes = {
  activityIndex: PropTypes.number.isRequired,
  useRouteMatch: PropTypes.func
};

export default ActivityRoutes;
