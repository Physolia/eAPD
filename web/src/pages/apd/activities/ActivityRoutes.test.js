import { shallow } from 'enzyme';
import React from 'react';

import ActivityRoutes from './ActivityRoutes';

const defaultProps = {
  activityIndex: 3,
  useRouteMatch: jest.fn().mockReturnValue({ path: '/your/current/path' })
};

const setup = (props = {}) => {
  return shallow(<ActivityRoutes {...defaultProps} {...props} />);
};

describe('<ActivityRoutes /> component', () => {
  it('renders as expected', () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
});
