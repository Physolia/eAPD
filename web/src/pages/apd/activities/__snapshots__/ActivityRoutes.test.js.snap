// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivityRoutes /> component renders as expected 1`] = `
<Switch>
  <Route
    path="/your/current/path/oms"
  >
    <Section
      id={null}
      resource={null}
    >
      <Connect(Outcomes)
        activityIndex={3}
      />
      <Connect(Milestone)
        activityIndex={3}
      />
    </Section>
  </Route>
  <Route
    path="/your/current/path/state-costs"
  >
    <Section
      id={null}
      resource={null}
    >
      <Costs
        activityIndex={3}
      />
    </Section>
  </Route>
  <Route
    path="/your/current/path/contractor-costs"
  >
    <Section
      id={null}
      resource={null}
    >
      <Connect(ContractorResources)
        activityIndex={3}
      />
    </Section>
  </Route>
  <Route
    path="/your/current/path/cost-allocation"
  >
    <Section
      id={null}
      resource={null}
    >
      <CostAllocation
        activityIndex={3}
      />
    </Section>
  </Route>
  <Route
    path="/your/current/path/ffp"
  >
    <Section
      id={null}
      resource={null}
    >
      <Connect(CostAllocateFFP)
        activityIndex={3}
      />
    </Section>
  </Route>
  <Route>
    <Section
      id={null}
      resource={null}
    >
      <Connect(ActivityOverview)
        activityIndex={3}
      />
      <Connect(StandardsAndConditions)
        activityIndex={3}
      />
    </Section>
  </Route>
</Switch>
`;
